wgsl/index.bs

```diff
@@ -528,14 +528,22 @@ An <dfn>identifier</dfn> is a kind of [=token=] used as a name.
 See [[#declaration-and-scope]] and [[#directives]].
 
 The form of an identifier is defined via pattern-matching, except that
-an identifier must not have the same spelling as a [=keyword=] or as a [=reserved word=].
+an identifier:
+* must not have the same spelling as a [=keyword=] or as a [=reserved word=].
+* must not be `_` (a single underscore)
+* must not start with two undercores.
 
 <div class='syntax' noexport='true'>
   <dfn for=syntax>ident</dfn> :
 
-    | `/[a-zA-Z][0-9a-zA-Z_]*/`
+    | `/([a-zA-Z_][0-9a-zA-Z][0-9a-zA-Z_]*)|([a-zA-Z][0-9a-zA-Z_]*)/`
```

Comment:
The logic here is:
 in the first case:
   - the first character to be a letter or underscore.
   - then number or letter
   - then any number of number, letter, underscore

In the second case:
  - letter
  - then any number, letter, underscore

So '_' is excluded because at best it would fall into the first case, but then require a second character.

Also '__' also starts out in the first case, but then is rejected because the second character must be number or letter

A single letter identifier is allowed, via the second case.